src/lib/Parser.hs

```diff
@@ -273,77 +274,36 @@ findImplicitImplicitArgNames typ = filter isLowerCaseName $ envNames $
     UIntLit  _ -> mempty
     UFloatLit _ -> mempty
 
-addImplicitImplicitArgs :: SrcPos -> Maybe UType -> UExpr -> (Maybe UType, UExpr)
-addImplicitImplicitArgs _ Nothing e = (Nothing, e)
-addImplicitImplicitArgs sourcePos (Just typ) ex =
-  let (ty', e') = foldr (addImplicitArg sourcePos) (typ, ex) implicitVars
+addImplicitImplicitArgs :: Maybe UType -> UExpr -> (Maybe UType, UExpr)
+addImplicitImplicitArgs Nothing e = (Nothing, e)
+addImplicitImplicitArgs (Just typ) ex =
+  let (ty', e') = foldr addImplicitArg (typ, ex) implicitVars
   in (Just ty', e')
   where
     implicitVars = findImplicitImplicitArgNames typ
 
-    addImplicitArg :: SrcPos -> Name -> (UType, UExpr) -> (UType, UExpr)
-    addImplicitArg pos v (ty, e) =
-      ( WithSrc (Just pos) $ UPi (Just uPat, uTyKind) ImplicitArrow ty
-      , WithSrc (Just pos) $ ULam (uPat, Just uTyKind) ImplicitArrow e)
+    addImplicitArg :: Name -> (UType, UExpr) -> (UType, UExpr)
+    addImplicitArg v (ty, e) =
+      ( ns $ UPi (Just uPat, uTyKind) ImplicitArrow ty
+      , ns $ ULam (uPat, Just uTyKind) ImplicitArrow e)
```

Comment:
We did have source info for those expressions before. Maybe consider keeping it?